pkg/kubelet/nodeinfo/nodeinfo_writter.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeinfo
+
+import (
+	"time"
+
+	"k8s.io/api/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+)
+
+type nodeInfoWritter interface {
+	SetRegistrationCompleted(value bool)
```

Comment:
These functions in particular seem indicative that the interface isn't at the right level...

For example, it is strange to me to have both SetNodeStatusFuncs() here, and SetNodeStatus() in the NodeInfo interface.  It ends up adding a few layers of indirection just so we can factor the `for` loop that iterates over the node status funcs out.  Additionally, the node status functions are static.  Most of these feel like they belong in a constructor, not as interface methods.

I would strongly prefer trying to add interfaces that are self-contained.  The pattern we seem to follow elsewhere in the kubelet is the NewFoo() + Start() pattern, which lets you add dependencies during either the construction time (New()), or later at Start().  If behavior (e.g. registration or status updates) is done at an interval, that is all defined inside the interface so we don't have to have triggering functions, and can encapsulate retry/backoff behavior inside the package.

Reply:
Edit: I just read the comment above that says you intend to further refactor the status update part.  I'd personally prefer if that were done in the same PR...  I'm primarily reviewing the quality of the interface since I assume the implementation isn't changing, and i'm not a fan of the current set of interfaces being added.

Reply:
@dashpole I agree with your thought and extract nodeStatusManager(It works fine in my local-cluster now). Could you give a very quick high level review? If these changes make sense, I'll clean these code further more and fix unit tests to pass all test.